src/wp-includes/rest-api/endpoints/class-wp-rest-abilities-list-controller.php

```diff
@@ -0,0 +1,335 @@
+		);
+
+		$context = $request->get_param( 'context' ) ?? 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$fields = $this->get_fields_for_response( $request );
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$links = array(
+				'self'       => array(
+					'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $ability->get_name() ) ),
+				),
+				'collection' => array(
+					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+				),
+			);
+
+			$links['run'] = array(
+				'href' => rest_url( sprintf( '%s/%s/%s/run', $this->namespace, $this->rest_base, $ability->get_name() ) ),
+			);
+
+			$response->add_links( $links );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the ability's schema, conforming to JSON Schema.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @return array<string, mixed> Item schema data.
+	 */
+	public function get_item_schema(): array {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'ability',
+			'type'       => 'object',
+			'properties' => array(
+				'name'          => array(
+					'description' => __( 'Unique identifier for the ability.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'label'         => array(
+					'description' => __( 'Display label for the ability.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'description'   => array(
+					'description' => __( 'Description of the ability.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'category'      => array(
+					'description' => __( 'Ability category this ability belongs to.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'input_schema'  => array(
+					'description' => __( 'JSON Schema for the ability input.' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'output_schema' => array(
+					'description' => __( 'JSON Schema for the ability output.' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'meta'          => array(
+					'description' => __( 'Meta information about the ability.' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+			),
+			'required'   => array( 'name', 'label', 'meta', 'description', 'category', 'input_schema', 'output_schema' ),
```

Comment:
```suggestion
```

We don't use top level required, and it doesn't really make sense on a read controller.